server/ControlPlane/Compute/Kubernetes/KubernetesRunCreator.cs

```diff
@@ -505,6 +511,61 @@ private async Task AddBufferProxySidecars(V1Pod jobPod, Run run, Dictionary<stri
         }
 
         await CreateObjectHandleAlreadyExists(() => _client.CoreV1.CreateNamespacedSecretAsync(buffersSecret, _k8sOptions.Namespace, cancellationToken: cancellationToken));
+
+        var secretRefreshTime = CalculateProactiveRefreshTimeFromNow(run.BufferAccessTtl!.Value);
+        await Repository.UpdateRunSecretRefreshTime(run.Id!.Value, secretRefreshTime, cancellationToken);
+    }
+
+    public async Task<bool> UpdateRunSecret(Run run, CancellationToken cancellationToken)
+    {
+        if (run.Job.Codespec is not JobCodespec jobCodespec)
+        {
+            return false;
+        }
+
+        var refreshed = await GetBufferMap(jobCodespec.Buffers, run.Job.Buffers!, run.BufferAccessTtl!, cancellationToken);
+
+        try
+        {
+            // Patch the secret with new buffer access URLs
+            var patchString = JsonSerializer.Serialize(new
+            {
+                stringData = refreshed.ToDictionary(p => p.Key, p => p.Value.sasUri.ToString()),
+            });
+            var patchBody = new V1Patch(patchString, V1Patch.PatchType.MergePatch);
+            var buffersSecretName = SecretNameFromRunId(run.Id!.Value);
+            await _client.CoreV1.PatchNamespacedSecretAsync(patchBody, buffersSecretName, _k8sOptions.Namespace, null, null, null, null, null, cancellationToken);
+        }
+        catch (HttpOperationException ex) when (ex.Response.StatusCode is HttpStatusCode.NotFound or HttpStatusCode.Conflict)
+        {
+            return false;
+        }
+
+        try
+        {
+            // Patch an annotation on the job pod metadata to trigger a refresh of the secret
```

Comment:
Is this to speed up the update? If we don't do this, is the secret change not reflected in the mounted volume?